### DISCUSSION TITLE
Default channel name format

### DISCUSSION THREAD

**Discussion by joannadiong** [MartinHeroux/spike2py#66] [General]
Quick question, I supplied a channel name in lowercase and by default, Trial generated an object that capitalised the first letter of the channel name by default. E.g, this was the setup:

```python
trial_info = TrialInfo(file=spike2_data, channels=['emgSO']') # using the name label of the Spike2 file's channel 
data = Trial(trial_info)
```

and this was the call to plot the values:

```python
data.Emgso.plot()
```

The line that generates the channel name doesn't seem to indicate this as default behaviour: 

https://github.com/MartinHeroux/spike2py/blob/c87874368fb8b40515124ccc962966e085914822/spike2py/trial.py#L34

Just curious if this behaviour was intended, as I thought the general preference was to keep variable names lower case? 

Thanks

**Comment by MartinHeroux** [comment]
The line you highlight is now where the channel names are given.

```python
 def _parse_trial_data(self):
        trial_data = self._import_trial_data()
        channel_names = list()
        for key, value in trial_data.items():
            channel_names.append((key.title(), value["ch_type"]))
            value["path_save_figures"] = self.info.path_save_figures
            value["trial_name"] = self.info.name
            value["subject_id"] = self.info.subject_id
            setattr(
                self,
                key.title(),
                CHANNEL_GENERATOR[value["ch_type"]](key, value),
            )
        self.channels = channel_names
```
Near the end, we set the class attribute that will be associated to the channel (`setattr`). Here, you can see that the name if the channel is converted to titlecase (`key.title()`).

I don't recall why I decided this convention, but it is the default behaviour.

**Reply by joannadiong** [reply]
Great, thanks for clarifying
